app/controllers/nonsipp/sharesdisposal/ReportedSharesDisposalListController.scala

```diff
@@ -295,47 +311,76 @@ object ReportedSharesDisposalListController {
-        "sharesDisposal.reportedSharesDisposalList.heading.plural"
-      )
-    }
+//    val (title, heading) = if (numberOfDisposals == 1) {
+//      ("sharesDisposal.reportedSharesDisposalList.title", "sharesDisposal.reportedSharesDisposalList.heading")
+//    } else {
+//      (
+//        "sharesDisposal.reportedSharesDisposalList.title.plural",
+//        "sharesDisposal.reportedSharesDisposalList.heading.plural"
+//      )
+//    }
+//
+//    val (titleView, headView) = if (numberOfDisposals == 1) {
+//      (
+//        "sharesDisposal.reportedSharesDisposalList.view.title",
+//        Message("sharesDisposal.reportedSharesDisposalList.view.heading", numberOfDisposals)
+//      )
+//    } else {
+//      (
+//        "sharesDisposal.reportedSharesDisposalList.view.title.plural",
+//        Message("sharesDisposal.reportedSharesDisposalList.view.heading.plural", numberOfDisposals)
+//      )
+//    }
 
-    val (titleView, headView) = if (numberOfDisposals == 1) {
-      (
-        "sharesDisposal.reportedSharesDisposalList.view.title",
-        Message("sharesDisposal.reportedSharesDisposalList.view.heading", numberOfDisposals)
-      )
-    } else {
-      (
-        "sharesDisposal.reportedSharesDisposalList.view.title.plural",
-        Message("sharesDisposal.reportedSharesDisposalList.view.heading.plural", numberOfDisposals)
-      )
+    val (title, heading) = ((mode, numberOfDisposals) match {
+      case (ViewOnlyMode, numberOfDisposals) if numberOfDisposals == 0 =>
+        (
+          "sharesDisposal.reportedSharesDisposalList.view.title",
+          "sharesDisposal.reportedSharesDisposalList.view.heading.none"
+        )
+
+      case (ViewOnlyMode, _) if numberOfDisposals > 1 =>
+        (
+          "sharesDisposal.reportedSharesDisposalList.view.title.plural",
+          "sharesDisposal.reportedSharesDisposalList.view.heading.plural"
+        )
+      case (ViewOnlyMode, numberOfDisposals) if numberOfDisposals > 1 =>
+        (
+          "sharesDisposal.reportedSharesDisposalList.view.title",
+          "sharesDisposal.reportedSharesDisposalList.view.heading"
+        )
+      case (ViewOnlyMode, numberOfDisposals) if numberOfDisposals > 1 =>
+        (
+          "sharesDisposal.reportedSharesDisposalList.title.plural",
+          "sharesDisposal.reportedSharesDisposalList.heading.plural"
+        )
+      case _ =>
+        (
+          "sharesDisposal.reportedSharesDisposalList.title",
+          "sharesDisposal.reportedSharesDisposalList.heading"
+        )
+    }) match {
+      case (title, heading) =>
+        (Message(title, numberOfDisposals), Message(heading, numberOfDisposals))
     }
 
-    val currentPage = if ((page - 1) * Constants.reportedSharesDisposalListSize >= numberOfDisposals) 1 else page
+//    val currentPage = if ((page - 1) * Constants.reportedSharesDisposalListSize >= numberOfDisposals) 1 else page
     val pagination = Pagination(
-      currentPage = currentPage,
+      currentPage = page,
```

Comment:
Is it save to take away the currentPage logic?

Reply:
I don't think so Kornel, It will take away currentPage logic, because we are following same for other pages.